### PR TITLE
[BACKLOG-4462] -- added ability to override getParameterDefinition by…

### PR DESCRIPTION
**Pull request by krivera-pentaho**
… providing a function to render function

### CODE REVIEW

**Comment by krivera-pentaho** [comment]
@annapopova1 @rfellows @scottyaslan @diogofscmariano @pamval @plagoa please review.

File: package-res/resources/web/prompting/api/OperationAPI.js [hunk 1]
```diff
@@ -48,22 +50,29 @@ define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/Param
...
-     * @name OperationAPI#update
-     * @method
-     * @param {String} xml XML string containing the parameter definition
-     */
-    this.update = function(xml) {
-      var paramDefn = this._parameterParser.parseParameterXml(xml);
-      this._getPromptPanel().refresh(paramDefn);
+      // Override of getParameterDefinition
+      this._promptPanel.getParameterDefinition = (function(promptPanel, refreshCallback) {
+        if (!getParameterXml) {
```
Comment (diogofscmariano) [inline package-res/resources/web/prompting/api/OperationAPI.js]: @krivera-pentaho we should also check if the getPameterXml is a function

File: package-res/resources/web/prompting/api/OperationAPI.js [hunk 2]
```diff
@@ -48,22 +50,29 @@ define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/Param
      * @method
      * @param {String} id HTML object id where the render the panel
      * @param {String} xml XML string containing the parameter definition
+     * @param {Function} getParameterXml The function called when the prompt panel needs to get the new parameter xml
      */
-    this.render = function(id, xml) {
+    this.render = function(id, xml, getParameterXml) {
```
Comment (diogofscmariano) [inline package-res/resources/web/prompting/api/OperationAPI.js]: @krivera-pentaho being the devil's advocate, why do we need to pass an xml and a func to get an xml?
Can't we just pass the callback and we call it in the first loop?

File: package-res/resources/web/prompting/api/OperationAPI.js (hunk 1 above)
Comment (krivera-pentaho) [inline package-res/resources/web/prompting/api/OperationAPI.js]: Ok, I will do that.

File: package-res/resources/web/prompting/api/OperationAPI.js (hunk 2 above)
Comment (krivera-pentaho) [inline package-res/resources/web/prompting/api/OperationAPI.js]: I agree with that as well. I will make the changes.

File: package-res/resources/web/prompting/api/promptApiSample.html [hunk 3]
```diff
@@ -13,366 +13,400 @@
...
+                var xml = $('#parameterXml').val();
+                promptApi.operation.render(id, xml);
 
-          promptApi.event.postInit(function() {
-            $('#eventsLogArea').append("<p>Executed Post Init event callback</p>");
-            $('#eventsLogArea').append("<p>qq</p>");
-          });
+                promptApi.event.postInit(function() {
+                    $('#eventsLogArea').append("<p>Executed Post Init event callback</p>");
+                    $('#eventsLogArea').append("<p>qq</p>");
```
Comment (annapopova1) [inline package-res/resources/web/prompting/api/promptApiSample.html]: @krivera-pentaho remove this line please ("<p>qq</p>"). It was test message.

**Pull request merged**